src/v/kafka/client/client.cc

```diff
@@ -310,6 +311,39 @@ ss::future<produce_response> client::produce_records(
         .throttle_time_ms{{std::chrono::milliseconds{0}}}}};
 }
 
+namespace {
+metadata_request copy_metadata_request(const metadata_request& req) {
```

Comment:
can we add this method as a `metadata_request::copy` method in `kafka/protocol/metadata.h`? also, wouldn't we want to validate the version range instead of its arity since the arity may stay the same but the copy logic still becomes outdated.